content/post/simulating-event-windows.md

```diff
@@ -0,0 +1,171 @@
++++
+title = "How to simulate event-driven windowing in Wallaroo"
+slug = "simulate-event-windowing-wallaroo"
+draft = true
+date = 2017-11-09T00:00:00Z
+tags = [
+    "wallaroo",
+    "windowing",
+    "partitioning"
+]
+categories = [
+    "Windowing"
+]
+description = "Creating event boundaries and processing cumulative state"
+author = "amosca"
++++
+
+Certain applications lend themselves to pure parallel computation better than others. In some cases we require to apply certain algorithms over a "window" in our data. This means that after we have complete a certain amount of processing (be it time, number of messages or some other arbitraty metric), we want to perform a special action for the data in that window. An example application of this could be producing stats for log files over a certain period of time. We may want to produce our stats in the form of periodic summarier (e.g. daily), in which case the computation in Wallaroo would have to know when it has received the last message for a particular day. We are working hard every day to support new use patterns, and this type of windowing can already be simulated without any native support in Wallaroo.
```

Comment:
"after we have complete" -> "after we have completed"
OR                                    -> "after we complete"

Reply:
"of periodic summarier" -> do you mean "of a periodic summary"?